.github/workflows/docker-build.yaml

```
@@ -0,0 +1,10 @@
+name: Build docker image
+
+on: pull_request
```

```suggestion
on:
  pull_request:
  push:
    branches:
      - main
```

This is to prepare for the comment above...